File: 프로그래머스/1/12926. 시저 암호/시저 암호.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s, int n) {
    string answer = "";
    
    for(int i=0; i<s.length(); i++){
        if(s[i] == ' ')
            continue;
        
        for(int j=0; j<n; j++){
            if(s[i] == 'z')
            {
                s[i] = 'a';
                continue;
            }
            if(s[i] == 'Z')
            {
                s[i] = 'A';
                continue;
            }
            
            s[i]++;
        }
    }
    
    answer = s;
    
    return answer;
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.js
function solution(s, n) {
    let alphabet = s.split("");
    let tmp = [];
    let answer = "";
    
    alphabet.forEach((el, idx) => {
        if (s.charCodeAt(idx) === 32) {
            tmp.push(String.fromCharCode(32));
            return;
        }
        
        // 아스키코드
        let charCode = s.charCodeAt(idx);
        
        if (charCode >= 65 && charCode <= 90)
            charCode = charCode + n > 90 ? charCode + n - 26 : charCode += n;

        if (charCode >= 97 && charCode <= 122)
            charCode = charCode + n > 122 ? charCode + n - 26 : charCode += n;
        
        tmp.push(String.fromCharCode(charCode));
    });
    
    answer = tmp.join("");
    
    return answer;
}